clients/client-glacier/GlacierClient.ts

```diff
@@ -249,6 +249,21 @@ export interface ClientDefaults
    * Fetch related hostname, signing name or signing region with given region.
    */
   regionInfoProvider?: RegionInfoProvider;
+
+  /**
+   * Browser only - chunked blobReader.
+   */
+  blobReader?: any;
+
+  /**
+   * Node only - createReadStream.
+   */
+  createReadStream?: any;
+
+  /**
+   * Node only - chunked streamReader.
+   */
+  streamReader?: any;
```

Comment:
When I think it more, I find it may looks nicer if we have only have one runtime specific config (like `contentChecksumGenerator`) instead of 3 here. In node `contentChecksumGenerator`, you can import `streamReader` package and use `fs.createReadStream` like usual. Whereas in browser one, you can import `blob` reader. This is nice because `blobReader` and `streamReader` is only used in calculating checksum now so this interface exposes less surface area.

In the middleware, you only need to apply the `contentChecksumGenerator` to the request in runtime-agnostic manner.

Reply:
Added body-checksum-node and body-checksum-browser to pull in these dependencies through one bodyChecksumGenerator